app/controllers/api/applications_controller.rb

```diff
@@ -1,6 +1,23 @@
 class Api::ApplicationsController < ActionController::API
   def index
-    fake_applications = File.read('lib/data/fake_applications.json')
-    render json: fake_applications
+    render json: applications
+  end
+
+  def show
+    if matching_application.present?
+      render json: matching_application
+    else
+      head :not_found
+    end
+  end
+
+private
+
+  def matching_application
+    applications.find { |app| app['id'] == params[:id] }
+  end
+
+  def applications
+    JSON.parse(File.read('lib/data/fake_applications.json'))
```

Comment:
I would take this abstraction one step further and create a PORO Application model in `app/models` with a `#all` and `#find` method. Add a spec for the dummy model too.

Reply:
I considered this to bring it into line with ActiveModel's API, to make it smoother to change across in the future, but I didn't for a few reasons:
- It presumes the shape of the next refactor, which feels weird
- It hides a very small amount of implementation detail
- We're very early on in the development, and things are very unstable

Ultimately, I thought it was just best to keep it simpler.

Do you feel strongly about this?

Reply:
Nope, just a recommendation based on expected direction of travel, but agree there’s an element of premature optimisation to it.  Happy to approve as is.